### PR TITLE
Fix bugs in findScripts

### PR DESCRIPTION
**Pull request by ofrobots**
* Code was buggy when `fileStats` had only one entry.
* Fix bug in `pathToRegExp` that was not considering '.' as literal.
* Make the code easier to unit test.
* More unit tests.

### CODE REVIEW

**Comment by ofrobots** [comment]
PTAL.

File: src/agent/v8debugapi.js [hunk 1]
```diff
@@ -585,4 +564,67 @@ module.exports.create = function(logger_, config_, jsFiles_, sourcemapper_) {
...
+  scriptPath = scriptPath.replace('.', '\\.');
+  if (path.sep !== '/') {
+    scriptPath = scriptPath.replace(new RegExp('\\\\', 'g'), '\\\\');
+  }
+  return new RegExp(scriptPath + '$');
+}
+
+/**
+ * Given an list available files and a script path to match, this function tries
+ * to resolve the script patch a (hopefully unique) match in the file list
```
Comment (matthewloring) [inline src/agent/v8debugapi.js]: `s/patch/path`

File: src/agent/v8debugapi.js [hunk 2]
```diff
@@ -585,4 +564,67 @@ module.exports.create = function(logger_, config_, jsFiles_, sourcemapper_) {
...
+ * irrespective of the script path. This can be useful because repo file paths
```
Comment (matthewloring) [inline src/agent/v8debugapi.js]: Not sure what you mean by `irrespective of the script path` here.

File: src/agent/v8debugapi.js (hunk 1 above)
Comment (ofrobots) [inline src/agent/v8debugapi.js]: done.

File: src/agent/v8debugapi.js (hunk 2 above)
Comment (ofrobots) [inline src/agent/v8debugapi.js]: done.

**Pull request merged**